src/json.ts

```diff
@@ -215,18 +216,20 @@ export const UnsafeJsonStruct: Struct<Json> = define('JSON', (json) =>
  * This struct sanitizes the value before validating it, so that it is safe to
  * use with untrusted input.
  */
-export const JsonStruct = coerce(UnsafeJsonStruct, any(), (value) => {
-  assertStruct(value, UnsafeJsonStruct);
-  return JSON.parse(
-    JSON.stringify(value, (propKey, propValue) => {
-      // Strip __proto__ and constructor properties to prevent prototype pollution.
-      if (propKey === '__proto__' || propKey === 'constructor') {
-        return undefined;
-      }
-      return propValue;
-    }),
-  );
-});
+export const JsonStruct = coerce(
```

Comment:
Can we add a test for the case specified in the description or similar?

Reply:
Done!